src/main/scala/org/broadinstitute/dsde/rawls/jobexec/MethodConfigResolver.scala

```diff
@@ -26,63 +28,69 @@ object MethodConfigResolver {
   }
 
+  private def getArrayResult(seq: Seq[AttributeValue]): Try[AttributeValueList] = {
+    // TODO figure out if this is the right approach to filtering nulls.
+    Success(AttributeValueList(seq.filter(v => v != null && v != AttributeNull)))
+  }
+
   /**
    * Evaluate expressions in a method config against a root entity.
-   * @return map from input name to a Try containing a resolved value or failure
+   * @return map from method config input name to a Try containing a resolved value or failure
    */
-  def resolveInputs(workspaceContext: WorkspaceContext, methodConfig: MethodConfiguration, entity: Entity, wdl: String, txn: RawlsTransaction): Map[String, Try[Attribute]] = {
-    val wdlInputs = NamespaceWithWorkflow.load(wdl, BackendType.LOCAL).workflow.inputs.map((w:WorkflowInput) => w.fqn -> w.wdlType).toMap
+  def evaluateMethodConfigInputs(workspaceContext: WorkspaceContext, methodConfig: MethodConfiguration, entity: Entity, txn: RawlsTransaction): Map[String, Try[Seq[AttributeValue]]] = {
     txn withGraph { graph =>
       val evaluator = new ExpressionEvaluator(new ExpressionParser)
       methodConfig.inputs.map {
         case (name, expression) => {
           val evaluated = evaluator.evalFinalAttribute(workspaceContext, entity.entityType, entity.name, expression.value)
-          // now look at expected WDL type to see if we should unpack a single value from the Seq[Any]
-          val unpacked = evaluated.flatMap(sequence => wdlInputs.get(name) match {
-              // TODO: Cromwell doesn't yet have compound types (e.g. Seq)
-              // TODO: so until it does just take the head
-              case Some(wdlType) => getSingleResult(sequence)
-              case None => getSingleResult(sequence)
-            }
-          )
-          (name, unpacked)
+          (name, evaluated)
         }
       }
     }
   }
 
   /**
-   * Get errors in both expression eval and WDL binding.
-   * @return map of from input name to error message (so if the map is empty, the config is valid)
+   * Unpack evaluated method config inputs using WDL.
+   * @return map from workflow input name to a Try containing an unpacked value or failure
    */
-  def getValidationErrors(workspaceContext: WorkspaceContext, methodConfig: MethodConfiguration, entity: Entity, wdl: String, txn: RawlsTransaction) = {
-    val configInputs = resolveInputs(workspaceContext, methodConfig, entity, wdl, txn)
-    NamespaceWithWorkflow.load(wdl, BackendType.LOCAL).workflow.inputs.map {
-      case w:WorkflowInput => {
-        val errorOption = configInputs.get(w.fqn) match {
-          case Some(Success(rawValue)) => {
-            // TODO: currently Cromwell's exposed type-checking is not very robust.
-            // TODO: once Cromwell has a validation endpoint, use that; until then, don't bother checking types.
-            None
+  def unpackWorkflowInputs(mcInputs: Map[String, Try[Seq[AttributeValue]]], wdl: String): Map[String, Try[Attribute]] = {
```

Comment:
It takes ```Seq[AttributeValue]``` because that's what we get from expression evaluation. It returns just ```Attribute``` because the output could be ```AttributeValueList```, which doesn't inherit from ```AttributeValue```. Perhaps the return should be ```Either```? After all, the return should definitely not be an ```AttributeReference```, for example.